src/components/FlexboxUtility/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { ReactNode } from "react";
+
+interface FlexboxComponentProps {
+    children: ReactNode;
+    flexDirection: "row" | "column";
+    className?: string;
+    gap?: number;
+    alignItems?: string;
+    justifyContent?: string;
+}
+
+type IFlexboxComponentProps = Omit<FlexboxComponentProps, "flexDirection">;
+
+const createFlexboxComponent = (
+    flexDirection: "row" | "column",
+    displayName: string
+) => {
+    const FlexboxUtility: React.FC<IFlexboxComponentProps> = ({
+        children,
+        className,
+        gap = 0,
+        alignItems = "stretch",
+        justifyContent = "flex-start",
+    }) => (
+        <div
+            className={className}
+            style={{
+                display: "flex",
+                flexDirection: flexDirection,
+                gap: `${gap}px`,
+                alignItems,
+                justifyContent,
+            }}
+        >
+            {children}
+        </div>
+    );
+
+    FlexboxUtility.displayName = displayName;
+    return FlexboxUtility;
+};
+
+const VerticalFlexbox = createFlexboxComponent("column", "VerticalFlexbox");
+const HorizontalFlexbox = createFlexboxComponent("row", "HorizontalFlexbox");
+
```

Comment:
These are tangential to the feature but I think will improve readability wherever we can use them, especially in highly nested JSX with many styling classes.

Reply:
curious: why this instead of antd `Flex`?

Reply:
My habit is to use my own JSX/CSS for this kind of thing, and was working backwards from that to a more DRY solution. Also taking inspiration from @ShrimpCryptid styling utilities in timelapse-colorizer.

Are we into antd `Flex`? We can go that way instead, I'm not married to this although it was fun to make.

Reply:
it basically has all the same props/capabilities, and is already imported. I'd say unless it doesn't do something that you need it to do, might as well use the library were already importing

Reply:
Bumping to 5 would be so fun, I'd love to do it. 😭